File: backend/app/schemas/onboarding.py
"""Schemas de onboarding."""

from pydantic import BaseModel, Field, field_validator


class OnboardingStatusResponse(BaseModel):
    """Estado actual del onboarding de un usuario."""

    onboarding_step: int | None = Field(
        default=None,
        description="Paso actual del wizard (1-5), None si está completado",
    )
    onboarding_completed: bool = Field(
        default=False,
        description="Indica si el usuario completó el wizard",
    )


class OnboardingStepUpdate(BaseModel):
    """Payload para actualizar el paso actual del onboarding."""

    step: int = Field(
        ge=1,
        le=5,
        description="Paso actual del wizard (1-5)",
    )

    @field_validator("step")
    def validate_step(cls, value: int) -> int:
        """Valida que el paso esté en el rango válido."""

        if value < 1 or value > 5:
            raise ValueError("step debe estar entre 1 y 5")
        return value


class OnboardingCompleteRequest(BaseModel):
    """Payload para marcar el onboarding como completado."""

    equipo_id: int | None = Field(
        default=None,
        description="ID del equipo creado durante el onboarding (None si el usuario saltó el wizard)",
    )
    api_key_id: int | None = Field(
        default=None,
        description="ID de la API key creada durante el onboarding (opcional)",
    )

File: frontend/src/api/onboarding.js
import client from "./client";

/**
 * Obtiene el estado actual del onboarding del usuario.
 * @returns {Promise<{onboarding_step: number|null, onboarding_completed: boolean}>}
 */
export async function getOnboardingStatus() {
  const { data } = await client.get("/onboarding/status");
  return data;
}

/**
 * Actualiza el paso actual del onboarding.
 * @param {number} step - Paso actual (1-5)
 * @returns {Promise<{onboarding_step: number|null, onboarding_completed: boolean}>}
 */
export async function updateOnboardingStep(step) {
  const { data } = await client.patch("/onboarding/step", { step });
  return data;
}

/**
 * Marca el onboarding como completado.
 * @param {Object} payload - Datos de los recursos creados
 * @param {number} payload.equipo_id - ID del equipo creado
 * @param {number} [payload.api_key_id] - ID de la API key creada (opcional)
 * @returns {Promise<void>}
 */
export async function completeOnboarding(payload) {
  await client.post("/onboarding/complete", payload);
}

/**
 * Resetea el onboarding para volver a realizar el wizard (solo admin).
 * @returns {Promise<void>}
 */
export async function resetOnboarding() {
  await client.post("/onboarding/reset");
}
